src/celttf/truetypefont.cpp

```diff
@@ -483,23 +497,34 @@ TextureFontPrivate::flush()
         indexes.push_back(index + 2);
     }
 
+    glBindBuffer(GL_ARRAY_BUFFER, m_vbo);
+    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, m_vio);
+
+    glBufferData(GL_ARRAY_BUFFER, sizeof(FontVertex) * MaxVertices, nullptr, GL_STREAM_DRAW);
+    glBufferData(GL_ELEMENT_ARRAY_BUFFER, sizeof(GLushort) * MaxIndices, nullptr, GL_STREAM_DRAW);
+
+    glBufferData(GL_ARRAY_BUFFER, sizeof(FontVertex) * MaxVertices, m_fontVertices.data(), GL_STREAM_DRAW);
+    glBufferData(GL_ELEMENT_ARRAY_BUFFER, sizeof(GLushort) * MaxIndices, indexes.data(), GL_STREAM_DRAW);
+
     glEnableVertexAttribArray(CelestiaGLProgram::VertexCoordAttributeIndex);
     glEnableVertexAttribArray(CelestiaGLProgram::TextureCoord0AttributeIndex);
     glVertexAttribPointer(CelestiaGLProgram::VertexCoordAttributeIndex,
                           2,
                           GL_FLOAT,
                           GL_FALSE,
                           sizeof(FontVertex),
-                          &m_fontVertices[0].x);
+                          reinterpret_cast<void*>(offsetof(FontVertex, x)));
```

Comment:
I think it is worth adding `static_assert(std::is_standard_layout_v<FontVertex>)` to the file to ensure that any future changes remain compatible with the use of `offsetof`.

Reply:
done